StudCity.Frontend/src/app/features/groupsSlice.js

```diff
@@ -0,0 +1,18 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const groupsSlice = createSlice({
+	name: 'groups',
+	initialState: {
+		viewMode: 'grid',
+	},
+	reducers: {
+		changeViewMode: (state, { payload }) => {
+			state.viewMode = payload;
+		},
+	},
+});
+export const { changeViewMode } = groupsSlice.actions;
+
+export const selectViewMode = (state) => state?.groups?.viewMode;
```

Comment:
you don't need to use optional chaining here, state will always have access to groups and viewMode, better wright: state.groups.viewMode